### DISCUSSION TITLE
[Chapter 05] Equipment 의 적절한 Type

### DISCUSSION THREAD

**Discussion by coalong** [cowaine/is-my-code-that-strange#41] [Chapter 04-05 (불변, 응집도)]
Page 84. 를 보면 아래와 같은 다이어그램이 있다.

<img width="300" alt="스크린샷 2024-06-22 오후 7 27 10" src="https://github.com/cowaine/is-my-code-that-strange/assets/71637466/96adf132-7c4b-4d81-a8be-3fe81142480b">

Equipment 는 데이터 클래스로 보여지는데 이런 경우 class 가 아닌 enum 으로 설계하면 어떨까 생각했습니다. 괜찮을까요?
class 와 enum 의 차이점과 어떤 경우 enum 을 사용하면 좋을지 궁금합니다.

**Comment by firewoody237** [comment]
Equipment는 장비인데, 상속구조도 있을 수 있다고 생각합니다. (아머 -> 스터디래더아머 -> 로우퀼리티드 스터디래더아머 등..)
그리고 장비마다 특수효과 같은 것도 있을 수 있는데, ENUM으로 구현하면 나중에 골치 아파지지 않을까도 싶네요..!
